torch/distributed/_tensor/dispatch.py

```diff
@@ -219,29 +225,21 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             suggested_input_schema.kwargs_schema,
             redistribute_with_schema=needs_redistribute,
         )
-
-        aten = torch.ops.aten
-        random_ops = [
-            aten.native_dropout.default,
-            aten.normal_.default,
-            aten.uniform_.default,
-        ]
-        # before running local op computation, check if op is random op
-        # for random ops, set RNG offset
-        assert isinstance(mesh, DeviceMesh)
-        if op_call in random_ops and is_rng_supported_mesh(mesh):
-            dtensor_arg = arg_list[0]
-            old_offset = _get_rng_offset(mesh)
-            set_pre_op_offset(dtensor_arg._spec)
-
         # run local op computation with potentially modified args/kwargs
         local_tensor_args = cast(Tuple[object, ...], local_tensor_args)
         local_tensor_kwargs = cast(Dict[str, object], local_tensor_kwargs)
-        local_results = op_call(*local_tensor_args, **local_tensor_kwargs)
-
-        # if op is a random op, adjust Philox RNG state to maintain synchronization
-        if op_call in random_ops and is_rng_supported_mesh(mesh):
-            set_post_op_offset(dtensor_arg._spec, old_offset)
+        # For DTensor random operator, run it within a parallel region
+        assert isinstance(mesh, DeviceMesh)
+        if _is_random_op(op_call) and is_rng_supported_mesh(mesh):
+            if not random._rng_tracker:
+                raise RuntimeError(
+                    "A CudaRNGStateTracker instance must be instantiated "
+                    "before executing a random op over a DTensor."
```

Comment:
Can we also elaborate more on how user can fix this situation? Maybe giving an example like "_rng_tracker = OffsetBasedRNGTracker()"?

Reply:
Yes! I'll append a guide to this error message.